components/brave_perf_predictor/browser/bandwidth_linreg_unittest.cc

```diff
@@ -0,0 +1,288 @@
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_perf_predictor/browser/bandwidth_linreg.h"
+
+#include "base/containers/flat_map.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_perf_predictor {
+
+TEST(BraveSavingsPredictorTest, FeatureArrayGetsPrediction) {
+  const std::array<double, feature_count> features{};
+  double result = LinregPredictVector(features);
+  EXPECT_NE(result, 0);
+}
+
+TEST(BraveSavingsPredictorTest, HandlesSpecificVectorExample) {
+  // This test needs to be updated for any change in the model
+  constexpr std::array<double, feature_count> sample = {
+      20, 129, 225, 225, 142, 925,    5, 34662, 3,  317818,  9,  1702888,
+      0,  0,   1,   324, 32,  238315, 9, 90131, 54, 2367498, 59, 2384138,
+      0,  1,   0,   0,   0,   0,      0, 0,     0,  1,       0,  0,
+      0,  0,   1,   0,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     1,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   1,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   1,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  1,   1,   1,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     0,  0,       0,  0,
+      0,  0,   0,   0,   0,   0,      0, 0,     0,  0};
+
+  const double result = LinregPredictVector(sample);
+  EXPECT_EQ(static_cast<int>(result / 1000),
+            794);  // Equal on the order of thousands
+}
+
+TEST(BraveSavingsPredictorTest, HandlesEmptyFeatureset) {
+  const base::flat_map<std::string, double> features{};
+  const double result = LinregPredictNamed(features);
+  const std::array<double, feature_count> features_array{};
+  const double array_result = LinregPredictVector(features_array);
+  EXPECT_EQ(result, array_result);
+}
+
+TEST(BraveSavingsPredictorTest, HandlesCompleteFeatureset) {
+  base::flat_map<std::string, double> features;
+  for (unsigned int i = 0; i < feature_count; i++) {
+    features[feature_sequence->at(i)] = 0;
+  }
+  const double result = LinregPredictNamed(features);
+  const std::array<double, feature_count> array_features{};
+  const double array_result = LinregPredictVector(array_features);
+  EXPECT_EQ(result, array_result);
+}
+
+TEST(BraveSavingsPredictorTest, HandesSpecificFeaturemapExample) {
+  // This test needs to be updated for any change in the model
```

Comment:
can't we use the same featuremap? In my experience people don't read these comments

Reply:
intention of the test is mostly to catch regressions from changing the set of features by accident and not propagating the changes elsewhere, hence the collection of fairly arbitrary collection of values. This block can be shortened though by skipping features with `0`